src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Cohort/WhenMappingWithTransferSenderRequestToViewModel.cs

```diff
@@ -0,0 +1,306 @@
+
+            fixture.Verify_AccountHashedId_IsMapped();
+        }
+
+        [Test]
+        public void Then_BacklinkUrl_IsMapped()
+        {
+            var fixture = new WhenMappingWithTransferSenderRequestToViewModelFixture();
+            fixture.Map();
+
+            fixture.Verify_BackLinkUrl_Is_Mapped();
+        }
+
+        [Test]
+        public void When_More_Than_One_TransferSender_Title_IsMapped()
+        {
+            var fixture = new WhenMappingWithTransferSenderRequestToViewModelFixture();
+            fixture.Map();
+
+            fixture.Verify_When_More_Than_One_TransferSender_Title_Is_Mapped();
+        }
+
+        [Test]
+        public void When_Only_One_TransferSender_Title_IsMapped()
+        {
+            var fixture = new WhenMappingWithTransferSenderRequestToViewModelFixture();
+            fixture.SetOnlyOneTransferSender();
+            fixture.Map();
+
+            fixture.Verify_When_One_TransferSender_Title_Is_Mapped();
+        }
+    }
+
+    public class WhenMappingWithTransferSenderRequestToViewModelFixture
+    {
+        public Mock<IEncodingService> EncodingService { get; set; }
+        public Mock<ICommitmentsApiClient> CommitmentsApiClient { get; set; }
+        public Mock<ILinkGenerator> LinkGenerator { get; set; }
+        public WithTransferSenderRequest WithTransferSenderRequest { get; set; }
+        public GetCohortsResponse GetCohortsResponse { get; set; }
+        public WithTransferSenderRequestViewModelMapper Mapper { get; set; }
+        public WithTransferSenderViewModel WithTransferSenderViewModel { get; set; }
+
+        public long AccountId => 1;
+
+        public string AccountHashedId => "1AccountHashedId";
+
+        public WhenMappingWithTransferSenderRequestToViewModelFixture()
+        {
+            EncodingService = new Mock<IEncodingService>();
+            CommitmentsApiClient = new Mock<ICommitmentsApiClient>();
+            LinkGenerator = new Mock<ILinkGenerator>();
+
+            WithTransferSenderRequest = new WithTransferSenderRequest() { AccountId = AccountId, AccountHashedId = AccountHashedId };
+            GetCohortsResponse = CreateGetCohortsResponse();
+
+            CommitmentsApiClient.Setup(c => c.GetCohorts(It.Is<GetCohortsRequest>(r => r.AccountId == AccountId), CancellationToken.None)).Returns(Task.FromResult(GetCohortsResponse));
+            EncodingService.Setup(x => x.Encode(It.IsAny<long>(), EncodingType.CohortReference)).Returns((long y, EncodingType z) => y + "_Encoded");
+            LinkGenerator.Setup(x => x.CommitmentsLink($"accounts/{AccountHashedId}/apprentices/cohorts")).Returns("BackLinkUrl");
+
+            Mapper = new WithTransferSenderRequestViewModelMapper(CommitmentsApiClient.Object, EncodingService.Object, LinkGenerator.Object);
+        }
+
+        public WhenMappingWithTransferSenderRequestToViewModelFixture Map()
+        {
+            WithTransferSenderViewModel = Mapper.Map(WithTransferSenderRequest).Result;
```

Comment:
This is an unusual name for the mapping result. With..... is usually used as a setup method. Would it not be better calling it MapResult or something slightly more explicit

Reply:
changed the name.